### PR TITLE
Updates to the Maintainers Guidelines and list of Maintainer Tasks

### PR DESCRIPTION
**Pull request by swcurran**
Signed-off-by: Stephen Curran <swcurran@gmail.com>

Updates the maintainers guidance document and adds a `Sample MAINTAINERS.md` file.

- Acknowledges that a MAINTAINERS.md file applies to a repo vs. a project.
- Talks about scope in terms of GitHub roles, or unrelated to GitHub Roles.
- Provides a sample Maintainers file.
- In the Sample Maintainers file, provides a list of tasks for open source software repository maintainers.

While I think this clarifies the Maintainer process and meaning, a key contribution is the tasks of an open source maintainer.

### CODE REVIEW

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [inline guidelines/MAINTAINERS-guidelines.md:28]: Why are we removing the other fields? The GitHub ID is not always enough to contact the person in question. You are removing the MUST to have chat ID or email. The company affiliation lets us know if we have diversity of maintainers for a project (i.e., that the project is not dependent on a single company to survive).

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [inline guidelines/MAINTAINERS-guidelines.md]:
```suggestion
All Hyperledger repositories **MUST** have a `MAINTAINERS` file (`MAINTAINERS.md` or `MAINTAINERS.rst`) at the top-level directory of the source code. The [SAMPLE-MAINTAINERS.md](SAMPLE-MAINTAINERS.md) can be used as a template by projects creating a new repository.
```

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [inline guidelines/MAINTAINERS-guidelines.md]:
```suggestion
Lists of active and emeritus maintainers **MUST** be included in the `MAINTAINERS` file.
```

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [inline guidelines/MAINTAINERS-guidelines.md]:
```suggestion
Changes to the maintainers lists **MUST** be made via Pull Requests. Once a new `MAINTAINERS` file is created or a PR changing the maintainer lists within the file is merged, a corresponding update to the affected GitHub Teams within the Hyperledger GitHub organization must be made. This is a manual process and the maintainers must ensure that it occurs.
```

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [inline guidelines/MAINTAINERS-guidelines.md]: This is a change from what was previously required. There were reasons for the other fields that I believe are still relevant. Please see my other comment for details.

File: guidelines/MAINTAINERS-guidelines.md
Comment (swcurran) [reply guidelines/MAINTAINERS-guidelines.md:28]: My experience with creating the table is that the information is hard to come by when initially populating the file when adding it to an existing repository.  It’s fine when starting from scratch, but painful when all you to start from is the GitHub Team.  I also think that some of the fields (e.g., LFID) are unnecessary.

AFAIK — the requirement to be able to contact the person is satisfied by the GitHub Id, so the extra methods were redundant.

I had never seen the company affiliation field previously (my bad 😄 ) and thought that would be painful to maintain. Is it being actively used for diversity reporting or is that asperational?

I’m open to adding some of the fields back, but recommend that we have a good reason for adding each one.

Comment (hartm) [reply guidelines/MAINTAINERS-guidelines.md:28]: I think the idea is that the maintainers themselves could populate this information, so I'm not sure it would be difficult to accumulate.  Any active maintainer should have no problems adding themselves to the table.

The LFID is useful for keeping track of people across companies and projects.  More importantly, it's useful for the people themselves, so they can show others what they've done even if they switch jobs or projects.

We are also actively using company affiliation to measure diversity--it is one of the most important contributor diversity metrics.  Lots of people ask us about the breakdowns of contributors by company, and it makes it difficult for us to answer this question if people contribute using their non-company email addresses (which is fine--we, and potential contributors, just want a way to measure company diversity).

File: guidelines/SAMPLE-MAINTAINERS.md
Comment (denyeart) [inline guidelines/SAMPLE-MAINTAINERS.md]: I'm not sure I like veto power in the template. I'm aware that Besu has a defined veto, but do any other Hyperledger projects? Is it typical in open source projects?  It seems like veto gives a whole lot of power to a single individual...

File: guidelines/SAMPLE-MAINTAINERS.md
Comment (denyeart) [inline guidelines/SAMPLE-MAINTAINERS.md]:
```suggestion
- The PR is authored by an existing maintainer or a Hyperledger staff member
```

Comment (denyeart) [reply guidelines/SAMPLE-MAINTAINERS.md]: Note - often times no existing maintainer wants to retire a colleague, it has been helpful to be able to ask a Hyperledger staff member to open a retirement PR. In fact, https://github.com/hyperledger/toc/blob/gh-pages/governing-documents/inactivity.md mentions it would be a TOC member or Hyperledger staff member, but I think it is acceptable for any project maintainer to do it.

Comment (ryjones) [reply guidelines/SAMPLE-MAINTAINERS.md]: Having done this many times by request, I agree with @denyeart

File: guidelines/MAINTAINERS-guidelines.md
Comment (ryjones) [inline guidelines/MAINTAINERS-guidelines.md]: Please make this MAINTAINERS.md _only_.

**Review by ryjones — DISMISSED** [review]
Please see wording change

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [reply guidelines/MAINTAINERS-guidelines.md]: @ryjones : Are you suggesting this as the change?
```suggestion
All Hyperledger repositories **MUST** have a `MAINTAINERS` file (i.e., `MAINTAINERS.md`) at the top-level directory of the source code. The [SAMPLE-MAINTAINERS.md](SAMPLE-MAINTAINERS.md) can be used as a template by projects creating a new repository.
```

Comment (ryjones) [reply guidelines/MAINTAINERS-guidelines.md]: @tkuhrt yes

File: guidelines/SAMPLE-MAINTAINERS.md
Comment (swcurran) [reply guidelines/SAMPLE-MAINTAINERS.md]: I took that from the example I had. I tried to clean up the language on it, but would be to soften it (objection, discussion, resolution) or remove it.  Perhaps easier to discuss?

File: guidelines/SAMPLE-MAINTAINERS.md
Comment (swcurran) [reply guidelines/SAMPLE-MAINTAINERS.md]: I think adding “or a Hyperledger Organization GitHub Administrator” would make sense.  I’ll try that.

File: guidelines/MAINTAINERS-guidelines.md
Comment (swcurran) [reply guidelines/MAINTAINERS-guidelines.md]: Instead, better to just change the generic “MAINTAINERS” references throughout the document to be “MAINTAINERS.md”.  I’ll do that and not have the parentheses in the text here.

File: guidelines/SAMPLE-MAINTAINERS.md
Comment (hartm) [reply guidelines/SAMPLE-MAINTAINERS.md]: It's not typical in other projects to my knowledge.  Some projects require supermajorities of maintainers to add another maintainer (2/3 or 3/4, for instance), but it's rare for a project to allow a single maintainer to block all other maintainers being added.

**Comment by swcurran** [comment]
Changes based on feedback applied, except about vetoes.  Perhaps a quick discussion at the TOC meeting?

File: guidelines/MAINTAINERS-guidelines.md
Comment (lehors) [inline guidelines/MAINTAINERS-guidelines.md]: Please make the link anchor meaningful:
```suggestion
like the [AnonCreds Methods Registry governance.md file](https://github.com/hyperledger/anoncreds-methods-registry/blob/main/registry/governance.md) might further define the duties of the maintainers.
```

**Review by ryjones — COMMENTED** [review]
lgtm modulo upcoming TOC call decisions

**Comment by swcurran** [comment]
Per the 2023.03.09 TOC Meeting, changes made:

- Veto removed from adding, removing maintainer in sample file.
- Added note for closing an unapproved PR to the maintainers file in sample file.
- Added Maintainer duty to sample `MAINTAINERS.md` about a reported security vulnerabilities. 
- Added note and example in sample `MAINTAINERS.md` file for referencing a common Project maintainers file.

Ready for another review.

File: guidelines/SAMPLE-MAINTAINERS.md
Comment (tkuhrt) [inline guidelines/SAMPLE-MAINTAINERS.md]:
```suggestion
- Review, respond, and act on any reported security vulnerabilities reported against the repository.
```

File: guidelines/MAINTAINERS-guidelines.md
Comment (tkuhrt) [reply guidelines/MAINTAINERS-guidelines.md:28]: With the re-addition of these columns, we still have an open item about whether an email address or chat ID should be required. Currently line 56 says that it is recommended.

**Comment by swcurran** [comment]
Updated the text in the Guidelines about what must be populated and why in the Maintainers table.

**Review by tkuhrt — APPROVED** [review]

**Review by lehors — APPROVED** [review]
LGTM. Thanks!

**Review by VRamakrishna — APPROVED** [review]

**Review by petermetz — APPROVED** [review]
LGTM

**Review by denyeart — APPROVED** [review]

**Comment by swcurran** [comment]
w00t!

**Pull request merged**
